insights/tests/combiners/test_rhel_for_edge.py

```diff
@@ -0,0 +1,110 @@
+from insights.parsers.installed_rpms import InstalledRpms
+from insights.parsers.cmdline import CmdLine
+from insights.parsers.systemd.unitfiles import ListUnits
+from insights.parsers.redhat_release import RedhatRelease
+from insights.combiners.rhel_for_edge import RhelForEdge
+from insights.combiners import rhel_for_edge
+from insights.tests import context_wrap
+import doctest
+
+CONTENT_REDHAT_RELEASE_RHEL = """
+Red Hat Enterprise Linux release 8.4 (Ootpa)
+""".strip()
+
+CONTENT_REDHAT_RELEASE_COREOS = """
+Red Hat Enterprise Linux CoreOS release 4.12
+""".strip()
+
+CMDLINE_RHEL = """
+BOOT_IMAGE=/vmlinuz-4.18.0-80.el8.x86_64 root=/dev/mapper/rootvg-rootlv ro crashkernel=184M rd.lvm.lv=rootvg/rootlv biosdevname=0 printk.time=1 numa=off audit=1 transparent_hugepage=always LANG=en_US.UTF-8
+""".strip()
+
+CMDLINE_EDGE = """
+BOOT_IMAGE=(hd0,msdos1)/ostree/rhel-323453435435234234232534534/vmlinuz-4.18.0-348.20.1.el8_5.x86_64 root=/dev/mapper/rhel-root ro crashkernel=auto rd.lvm.lv=rhel/root rd.lvm.lv=rhel/swap ostree=/ostree/boot.1/rhel/08987978979/0 rhgb quiet LANG=en_US.UTF-8
+""".strip()
+
+CONTENT_SYSTEMCTL_LIST_UNITS_NO_AUTOMATED = """
+postgresql.service                                                                               loaded active running   PostgreSQL database server
+""".strip()
+
+CONTENT_SYSTEMCTL_LIST_UNITS_AUTOMATED = """
+postgresql.service                                                                               loaded active running   PostgreSQL database server
+rhcd.service                                                                                     loaded active running   Red Hat connector daemon
+""".strip()
+
+CONTENT_INSTALLED_RPMS_RHEL = """
+kernel-4.18.0-305.19.1.el8_4.x86_64
+""".strip()
+
+CONTENT_INSTALLED_RPMS_EDGE = """
+kernel-4.18.0-305.19.1.el8_4.x86_64
+rpm-ostree-2021.5-2.el8.x86_64
+""".strip()
+
+
+def test_rhel_for_edge_true_1():
```

Comment:
You have five tests for testing the same structure/behaviour. I think that usage of `@pytest.mark.parametrize` would be helpful here. (I also think the variable names could be shorter, e.g. without the `CONTENT_` prefix.)

All of your five tests can be written as follows:
```
@pytest.mark.parametrize("rpms, cmdline, systemctl, rhel, is_edge, is_automated", [
    (CONTENT_INSTALLED_RPMS_EDGE, CMDLINE_EDGE, CONTENT_SYSTEMCTL_LIST_UNITS_NO_AUTOMATED,
     CONTENT_REDHAT_RELEASE_RHEL, True, False),
    (CONTENT_INSTALLED_RPMS_EDGE, CMDLINE_EDGE, CONTENT_SYSTEMCTL_LIST_UNITS_AUTOMATED,
     CONTENT_REDHAT_RELEASE_RHEL, True, True),
    (CONTENT_INSTALLED_RPMS_RHEL, CMDLINE_EDGE, CONTENT_SYSTEMCTL_LIST_UNITS_AUTOMATED,
     CONTENT_REDHAT_RELEASE_RHEL, False, False),
    (CONTENT_INSTALLED_RPMS_EDGE, CMDLINE_RHEL, CONTENT_SYSTEMCTL_LIST_UNITS_AUTOMATED,
     CONTENT_REDHAT_RELEASE_RHEL, False, False),
    (CONTENT_INSTALLED_RPMS_EDGE, CMDLINE_EDGE, CONTENT_SYSTEMCTL_LIST_UNITS_AUTOMATED,
     CONTENT_REDHAT_RELEASE_COREOS, False, False)
])
def test_rhel_for_edge(rpms, cmdline, systemctl, rhel, is_edge, is_automated):
    install_rpms = InstalledRpms(context_wrap(rpms))
    cmdline = CmdLine(context_wrap(cmdline))
    list_units = ListUnits(context_wrap(systemctl))
    redhat_release = RedhatRelease(context_wrap(rhel))

    result = RhelForEdge(install_rpms, cmdline, list_units, redhat_release)
    assert result.is_edge is is_edge
    assert result.is_automated is is_automated
```

Let me know if you have any questions regarding the example I wrote.

Reply:
@jobselko - I think current test format is OK, you can raise a PR to update the test format which you prefer after this PR gets merged.